dolphin/api/v1/storages.py

```diff
@@ -154,6 +156,18 @@ def sync_all(self, req):
                   format(len(storages)))
 
         for storage in storages:
+            if storage[constants.DB.DEVICE_SYNC_STATUS] != \
+                    constants.SyncStatus.SYNCED:
+                LOG.warn('sync task is running for %s' % storage['id'])
+                continue
+            # Set all bits of sync_status to SYNCING and start sync tasks
+            storage[constants.DB.DEVICE_SYNC_STATUS] = utils.set_bits(
+                storage[constants.DB.DEVICE_SYNC_STATUS],
+                0,
+                len(constants.ResourceType) - 1,
+                constants.SyncStatus.SYNCING)
+            db.storage_update(ctxt, storage['id'], storage)
```

Comment:
Think about the scenario that two sync_all rquest arived at the same time, let's say request_A and request_B, handle steps as below:
1. request_A get storage info first, and then request_B get storage info too,
2. request_A check sync_status is not syncing so update it to be syncing, 
3. request_B check sync_status is not syncing so it will update the sync status again, which is not correct.
So read and update sync_status need lock.

Reply:
Done